### PR TITLE
[ssu] Introduce SsuFeatureModel. Fixes MER#1367

### PR DESCRIPTION
**Pull request by jpetrell**

### CODE REVIEW

File: tests/ut_featuremodel/featuremodeltest.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+void FeatureModelTest::cleanupTestCase(){
+
+}
+
+void FeatureModelTest::testFeatures(){
+    SsuFeatureModel featureModel(0, ":/testdata");
+
+    // Features with no name or version were skipped (feature-noname.ini, feature-noversion.ini)
+    QCOMPARE(featureModel.count(), 2);
```
Comment (saukko) [inline tests/ut_featuremodel/featuremodeltest.cpp]: I wonder if we should skip only entries without name, but when version is missing just show name in the lists?

**Comment by jpetrell** [comment]
Updated to accept features without version field.

**Comment by saukko** [comment]
:+1:

**Comment by jpetrell** [comment]
Added Qt5Qml build requires to ssu-declarative.

**Comment by pvuorela** [comment]
lgtm.

**Comment by jpetrell** [comment]
Fixed Raine's findings.

**Comment by jpetrell** [comment]
Fixed unnecessary forward declaration.

**Comment by mikkoharju** [comment]
lgtm too

**Comment by rainemak** [comment]
LGTM too :+1:

**Pull request merged**